crates/volta-core/src/command.rs

```diff
@@ -28,3 +31,52 @@ cfg_if! {
         }
     }
 }
+
+/// Rebuild command against given PATH
+///
+/// On Windows, we need to explicitly use an absolute path to the executable,
+/// otherwise the executable will not be located properly, even if we've set the PATH.
+/// see: https://github.com/rust-lang/rust/issues/37519
+///
+/// This function will try to find the executable in the given path and rebuild
+/// the command with the absolute path to the executable.
+pub fn rebuild_command<S: AsRef<OsStr>>(command: Command, path: S) -> Fallible<Command> {
+    debug!("PATH: {}", path.as_ref().to_string_lossy());
+
+    #[cfg(not(windows))]
+    {
+        let mut command = command;
+        command.env("PATH", path.as_ref());
+        Ok(command)
+    }
+
+    #[cfg(windows)]
+    {
+        let args = command.get_args().collect::<Vec<_>>();
+        //          cmd /c <name> [...other]
+        // args_idx     0  1      2..
+        let name = args.get(1).expect("should have the name");
+
+        if let Ok(mut paths) = which::which_in_global(name, Some(&path)) {
```

Comment:
Given that the trigger for this (as I understand it) is the `cmd /c` implementation searching the CWD for the binary, even when it's not explicitly on the `Path`, would it be unreasonable to gate this reformulation of the `Command` on that, rather than doing it every time on Windows?

It's relatively minor, but the `which_in_global` is going to necessarily do a decent chunk of File I/O for every command run, which seems excessive for something on the hot path.